File: src/prod/src/data/txnreplicator/common/TransactionState.cpp
// ------------------------------------------------------------
// Copyright (c) Microsoft Corporation.  All rights reserved.
// Licensed under the MIT License (MIT). See License.txt in the repo root for license information.
// ------------------------------------------------------------

#include "stdafx.h"

using namespace ktl;

namespace TxnReplicator
{
    namespace TransactionState
    {
        void WriteToTextWriter(Common::TextWriter & w, Enum const & val)
        {
            switch (val)
            {
            case Active:
                w << L"Active"; return;
            case Reading:
                w << L"Reading"; return;
            case Committing:
                w << L"Committing"; return;
            case Aborting:
                w << L"Aborting"; return;
            case Committed:
                w << L"Committed"; return;
            case Aborted:
                w << L"Aborted"; return;
            case Faulted:
                w << L"Faulted"; return;
            default:
                Common::Assert::CodingError("Unknown TransactionState:{0} in Replicator", val);
            }
        }

        ENUM_STRUCTURED_TRACE(TransactionState, Active, LastValidEnum)
    }
}

File: src/prod/src/managed/Microsoft.ServiceFabric.Data.Impl/Replicator/TransactionState.cs
// ------------------------------------------------------------
// Copyright (c) Microsoft Corporation.  All rights reserved.
// Licensed under the MIT License (MIT). See License.txt in the repo root for license information.
// ------------------------------------------------------------

namespace Microsoft.ServiceFabric.Replicator
{
    /// <summary>
    /// Transaction State
    /// </summary>
    public enum TransactionState : int
    {
        /// <summary>
        /// Active
        /// </summary>
        Active = 0,
            
        /// <summary>
        /// A read is ongoing
        /// </summary>
        Reading = 1,
        
        /// <summary>
        /// Transaction is being committed
        /// </summary>
        Committing = 2,

        /// <summary>
        /// Transaction is being aborted
        /// </summary>
        Aborting = 3,

        /// <summary>
        /// Transaction is committed
        /// </summary>
        Committed = 4,

        /// <summary>
        /// Transaction is aborted
        /// </summary>
        Aborted = 5,
        
        /// <summary>
        /// Transaction failed to commit/abort due to an exception
        /// </summary>
        Faulted = 6
    }
}